src/screens/IndexScreen.js

```diff
@@ -107,37 +114,93 @@ export const IndexScreen = ({ navigation, route }) => {
+        setQueryVariables((prevQueryVariables) => {
+          // remove a refetch key if present, which was necessary for the "- Alle -" selection
+          delete prevQueryVariables.refetch;
 
-        setQueryVariables({ ...queryVariables, ...additionalQueryVariables });
+          return {
+            ...prevQueryVariables,
+            ...getAdditionalQueryVariables(query, selectedValue, excludeDataProviderIds)
+          };
+        });
       } else {
+        if (
+          Object.prototype.hasOwnProperty.call(queryVariables, [
+            keyForSelectedValueByQuery?.[query]
+          ])
+        ) {
+          setQueryVariables((prevQueryVariables) => {
+            // remove the filter key for the specific query if present, when selecting "- Alle -"
+            delete prevQueryVariables[keyForSelectedValueByQuery[query]];
+
+            // need to spread the prior `queryVariables` into a new object with additional
+            // refetch key to force the Query component to update the data, otherwise it is
+            // not fired somehow because the state variable wouldn't change
+            return { ...prevQueryVariables, refetch: true };
+          });
+        }
+      }
+    },
+    [query, queryVariables, excludeDataProviderIds]
+  );
+
+  const updateListDataByDailySwitch = useCallback(() => {
+    // update switch state as well
+    setFilterByDailyEvents((oldSwitchValue) => {
+      // if `oldSwitchValue` was false, we now activate the daily filter and set a date range
+      if (!oldSwitchValue) {
         setQueryVariables((prevQueryVariables) => {
-          // remove the filter key for the specific query, when selecting "- Alle -"
-          delete prevQueryVariables[keyForSelectedValueByQuery[query]];
-          // need to spread the `prevQueryVariables` into a new object with additional refetch key
+          // remove a refetch key if present, which was necessary for unselecting daily events
+          delete prevQueryVariables.refetchDate;
+
+          // add the filter key for the specific query, when filtering for daily events
+          return { ...prevQueryVariables, dateRange: [currentDate, currentDate] };
+        });
+      } else {
+        setQueryVariables((prevQueryVariables) => {
+          // remove the filter key for the specific query, when unselecting daily events
+          delete prevQueryVariables.dateRange;
+
+          // need to spread the `prevQueryVariables` into a new object with additional refetchDate key
           // to force the Query component to update the data, otherwise it is not fired somehow
           // because the state variable wouldn't change
-          return { ...prevQueryVariables, refetch: true };
+          return { ...prevQueryVariables, refetchDate: true };
         });
       }
-    },
-    [excludeDataProviderIds, setQueryVariables, query, queryVariables]
-  );
+
+      return !oldSwitchValue;
+    });
+  }, [filterByDailyEvents, queryVariables]);
```

Comment:
When I go to the "Veranstaltungen" page from the main page, today's events are displayed. That's because the filter is switched on. However, if I go directly to the "Veranstaltungen" page from the "Meldungen" or "Nachrichten" page, all events are displayed, even though the filter is activated.

Reply:
thanks for this very good review catch! fixed with bb5a256